graph/patterns/navigation-property.md

```diff
@@ -0,0 +1,185 @@
+
+For both many-to-one and one-to-many relationships in order to retrieve information from both resources on both sides of the relationship requires at minimum two requests.
+
+Requiring two round trips to access this information is inefficient for some applications and the lack of a formally described relationship limits the ability for tooling to take advantage of the relationship to improve developer experience.
+
+## Solution
+--------
+
+Navigation properties are an [OData convention](https://docs.microsoft.com/en-us/odata/webapi/model-builder-untyped#navigation-property) that allows an API designer to describe a special kind of property in a model that references an related entity. In the HTTP API this property name translates to a path segment that can be appended to the URL of the primary resource in order to access a representation of the related resource. This prevents the client from needing to know any additional information on how to construct the URL to the related resource and the client does not need to retrieve the primary resource if it is only interested in the related resource.  It is the responsibility of the API implementation to determine the Id of the related resource and return the representation of the related entity.
+
+e.g. /user/{userId}/manager  # many-to-one relationship
+     /user/{userId}/messages # one-to-many relationship
+
+Additionally, using the OData Expand query parameter, related entities can be transcluded into the primary entity so both can be retrieved in a single round trip.
+
+## Issues and Considerations
+-------------------------
+
+In the current Microsoft Graph implementation, support for navigation properties is limited to entities within the same backend service or the user entity.  
+ 
+Navigation properties defined within an entity are not returned when retreiving the representation of an entity.  
+
+Implementing support for accessing the "$ref" of a navigation property allows a caller to return just the URL of related resource. e.g. `/user/23/manager/$ref`. This is useful when a client wishes to identity the related resource but doesn't need all of its properties.
+
+## When to Use this Pattern
+------------------------
+
+### "has a" relationships
+
+The use of navigation properties is preferred over including an Id field to reference the related entity in a many-to-one relationship.  Id values require a client to make two round trips to retrieve the details of a related entity.  With a navigation property a client can retrieve a related entity in a single round trip. 
+
+Navigation properties are also useful when clients sometimes want to retrieve both the primary entity and the related entity in a single round trip.  The `expand` query parameter makes this possible.
+
+### "parent-child" relationships
+
```

Comment:
IMO it's better to call containment relationships since parent-child in OOP is usually inheritance so all c# devs will be confused.

Reply:
I'm OK with parent-child relationship.  I also like to point out the lifecycle implications.  The child entities do not have an independent lifecycle - when the parent is deleted, it's children are too.

Reply:
@dkershaw10   Except for when they don't.  You could have a one-to-many relationship where deleting the parent doesn't delete the children.  OneDrive Bundles is an example. Terms in TermSet is another. Perhaps we could argue that "parent-child" infers containsTarget=true, however, we have even stated that containsTarget doesn't define lifetime completely.  A non-contained collection could cause its children to be deleted when it is deleted.

I have tried to avoid bringing up the topic of lifetime because I think it deserves a section all to itself, but I am torn about the right way to explain this.